devops/girder/plugins/AnnotationPlugin/upenncontrast_annotation/server/helpers/connections.py

```diff
@@ -13,7 +13,7 @@ def pointToPointDistance(coord1, coord2):
         Number: Squared distance between point1 and point2
```

Comment:
the documentation says that a z coord is expected, code must be consistent with documentation

Reply:
In the functions simpleCentroid and pointToPointDistance, the z coordinate should be ignored, as it has no meaning, I forgot to update the doc.
It was declared because GeoJS sometimes output a z coordinate (and sometimes not, see [GeoJS doc](https://opengeoscience.github.io/geojs/apidocs/geo.html#.geoPosition)).
But in our case it corresponds to nothing, not even the z coordinate of the tile or the channel.
I think it is error prone to keep a variable that shouldn't be used in the code. This is what caused #304.
Whe could eventually mention it in the interface declaration :
```
export interface IGeoJSPoint {
  x: number;
  y: number;
  z?: number; // Artifact from GeoJS, unused
}
```

Reply:
> Please write "Closes #304" instead of `[issue 304](https://github.com/Kitware/UPennContrast/issues/304)`
> 
> I do not think removing `z` is a good idea. Some alternatives:
> 
> * make sure it is never `undefined`
> * if undefined, consider it to be 0
> * check in functions if you have xy or xyz

@manthey WDYT ?

Reply:
Actually we store in the database a 'Location' entry, for each annotation, defines as {"XY": n, "Z": m}.
When looking for the _closestAnnotation_, we filter annotations on their locations, so a "Z" value is already taken into account. So I don't think it is an issue to remove it from distance computation